packages/search/tests/unit/services/navi-report-search-provider-test.js

```diff
@@ -0,0 +1,44 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import setupMirage from 'ember-cli-mirage/test-support/setup-mirage';
+// import { settled } from '@ember/test-helpers';
+
+module('Unit | Service | navi-report-search-provider', function(hooks) {
+  setupTest(hooks);
+  setupMirage(hooks);
+
+  hooks.beforeEach(async function() {
+    // Load metadata needed for request fragment
+    await this.owner.lookup('service:bard-metadata').loadMetadata();
+  });
+
+  // Replace this with your real tests.
+  test('it exists', function(assert) {
+    const service = this.owner.lookup('service:navi-search/navi-report-search-provider');
+    assert.ok(service);
+  });
+
+  test('search by user', async function(assert) {
+    const service = this.owner.lookup('service:navi-search/navi-report-search-provider');
+    const results = await service.search({ title: 'Hyrule', request: 'clicks' }, 'navi_user');
```

Comment:
what if you pass in a key that is not in the request object?

Reply:
or duplicate keys? or if you pass in something that's not an object? like undefined, or the integer `1`